### PR TITLE
Move teams state out of reader

### PR DESCRIPTION
**Pull request by sgomes**
Teams state, which is currently a part of Reader state, is now used in layout, after #48875.

Since Reader state is so large, and since teams state is entirely independent of the rest of Reader, it makes sense for both performance and organisational reasons to extract it into its own top-level piece of state.

This should effectively revert the 7KB (gzipped) increase in `entry-main` size that #48875 introduced.

#### Changes proposed in this Pull Request

* Move `teams` state out of `reader`

#### Testing instructions

* Install [Redux DevTools](https://chrome.google.com/webstore/detail/redux-devtools/lmhkpmbekcpmknklioeibfkpmmfibljd?hl=en).
* Open DevTools and switch to the Redux tab.
* Open `/home` on the live branch or your local copy of this branch.
* Click `State` on the top right to ensure you see all state, and not just the diff for the last action: 
image
* Analyse the tree and verify that there's no `reader` key, even if you expand the list of keys. This indicates that the reader state hasn't been loaded.
* Analyse the tree and verify that there is a `teams` key. This indicates that the teams state has been loaded.
* Verify that navigation displays and works correctly, and that the correct navigation is chosen based on the type of user

### CODE REVIEW

**Review by cpapazoglou — APPROVED** [review]
I can verify that:
- when loading `/home` state doesn't include a `reader` key but includes a `team` key
- when loading `/read` the `reader` key is added
- everything works as expected when navigating both in `/home` and `/reader`

**Comment by cpapazoglou** [comment]
Thanks @sgomes for working on this, I should have noted this in my PR..

File: client/state/teams/actions.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+/**
+ * Internal dependencies
+ */
+import { TEAMS_REQUEST } from 'calypso/state/teams/action-types';
+
+import 'calypso/state/data-layer/wpcom/read/teams';
+
+import 'calypso/state/teams/init';
```
Comment (cpapazoglou) [inline client/state/teams/actions.js:8]: Perhaps it is a good time changing this too? It will require a wpcom change also to `read/teams`

Comment (sgomes) [reply client/state/teams/actions.js:8]: This line (L8) is unrelated to wpcom or anything outside of Calypso. It's an artifact of how we handle [modularised state](https://github.com/Automattic/wp-calypso/blob/trunk/docs/modularized-state.md).

**Comment by sgomes** [comment]
Thank you for the review, @cpapazoglou! 👍

File: client/state/teams/actions.js (hunk 1 above)
Comment (cpapazoglou) [reply client/state/teams/actions.js:8]: Sorry, I meant `import 'calypso/state/data-layer/wpcom/read/teams';` on line (6).

Comment (sgomes) [reply client/state/teams/actions.js:8]: Oh, I see! Yes, the paths could certainly be standardised at some point 👍

**Pull request merged**